extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/K8sTaskId.java

```diff
@@ -41,8 +38,7 @@ public K8sTaskId(String taskId)
   {
     this.originalTaskId = taskId;
     // replace all the ": - . _" to "", try to reduce the length of pod name and meet pod naming specifications 64 characters.
-    this.k8sTaskId = StringUtils.left(RegExUtils.replaceAll(taskId, "[^a-zA-Z0-9\\\\s]", "")
-                                                .toLowerCase(Locale.ENGLISH), 63);
+    this.k8sTaskId = KubernetesStringUtils.parseStringToK8sLabel(taskId);
```

Comment:
I know the existing code has this too but this truncation could cause loss of uniqueness. Is that not a concern? I assume it should be okay if they are just to be used as labels.

Reply:
it wouldn't be a concern with labels, but this is also what we name the job so there may be some concern there